### PR TITLE
fix parse pairs

### PR DESCRIPTION
**Pull request by xilabao**
**What this PR does / why we need it**:
add check to `-`
```
# kubectl label pod foo -               
error: at least one label update is required
See 'kubectl label -h' for help and examples.
```

**Which issue this PR fixes**: 

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by fabianofranz** [comment]
@k8s-bot ok to test

**Comment by xilabao** [comment]
@k8s-bot pull-kubernetes-unit test this
@k8s-bot pull-kubernetes-e2e-gce-etcd3 test this

**Comment by fabianofranz** [comment]
@k8s-bot pull-kubernetes-e2e-gce-etcd3 test this

File: pkg/kubectl/cmd/util/helpers.go [hunk 1]
```diff
@@ -620,7 +620,7 @@ func AddInclude3rdPartyVarFlags(cmd *cobra.Command, include3rdParty *bool) {
 func GetResourcesAndPairs(args []string, pairType string) (resources []string, pairArgs []string, err error) {
 	foundPair := false
 	for _, s := range args {
-		nonResource := strings.Contains(s, "=") || strings.HasSuffix(s, "-")
+		nonResource := strings.Contains(s, "=") || (strings.HasSuffix(s, "-") && s != "-")
```
Comment (soltysh) [inline pkg/kubectl/cmd/util/helpers.go]: If you're fixing this for hyphen (`-`) can you also add similar to check for the equal sign (`=`). I'm guessing the same error will appear if you use it.

File: pkg/kubectl/cmd/util/helpers.go [hunk 2]
```diff
@@ -657,7 +657,7 @@ func ParsePairs(pairArgs []string, pairType string, supportRemove bool) (newPair
 			} else {
 				newPairs[parts[0]] = parts[1]
 			}
-		} else if supportRemove && strings.HasSuffix(pairArg, "-") {
+		} else if supportRemove && strings.HasSuffix(pairArg, "-") && pairArg != "-" {
```
Comment (soltysh) [inline pkg/kubectl/cmd/util/helpers.go]: Same here, although applied to a previous if.

**Review by soltysh — CHANGES_REQUESTED** [review]
A few nits.

**Comment by xilabao** [comment]
@k8s-bot pull-kubernetes-federation-e2e-gce test this

**Comment by xilabao** [comment]
@k8s-bot pull-kubernetes-federation-e2e-gce test this

**Comment by xilabao** [comment]
@k8s-bot pull-kubernetes-federation-e2e-gce test this

**Comment by xilabao** [comment]
seems I cann't test it. @soltysh PTAL

**Comment by fabianofranz** [comment]
@k8s-bot pull-kubernetes-federation-e2e-gce test this

**Comment by xilabao** [comment]
@soltysh kindly ping. ptal

**Comment by xilabao** [comment]
@soltysh ptal

**Review by soltysh — APPROVED** [review]
lgtm

**Comment by soltysh** [comment]
Sorry, was on PTO.
/lgtm

**Comment by soltysh** [comment]
@fabianofranz mind approving?

**Comment by fabianofranz** [comment]
Needs additional approvers.

**Pull request merged**